src/common/BackTrace.cc

```diff
@@ -2,29 +2,33 @@
 // vim: ts=8 sw=2 smarttab
 
 #include <ostream>
-#include <cxxabi.h>
 #include <string.h>
 
+#include <boost/stacktrace.hpp>
+
 #include "BackTrace.h"
 #include "common/version.h"
 #include "common/Formatter.h"
 
 namespace ceph {
 
+BackTrace::BackTrace(std::size_t s)
+  : bt{s, max}
+{}
+
 void BackTrace::print(std::ostream& out) const
 {
   out << " " << pretty_version_to_str() << std::endl;
-  for (size_t i = skip; i < size; i++) {
-    out << " " << (i-skip+1) << ": " << demangle(strings[i]) << std::endl;
+  for (size_t i = 0; i < bt.size(); i++) {
```

Comment:
why is `skip` skipped?

Reply:
I think it's just delegated outside, to the `stacktrace`. Before the change `skip` was set basing the value passed to the constructor:

```diff
-   explicit BackTrace(int s) : skip(s) {
```

which is now supplied to `bt`

```diff
+BackTrace::BackTrace(int s)
+  : bt{s, max}
+{}
```